bigchaindb/upsert_validator/validator_election_vote.py

```diff
@@ -0,0 +1,65 @@
+import base58
+
+from bigchaindb.common.transaction import Transaction
+from bigchaindb.common.schema import (_validate_schema,
+                                      TX_SCHEMA_COMMON,
+                                      TX_SCHEMA_TRANSFER,
+                                      TX_SCHEMA_VALIDATOR_ELECTION_VOTE)
+
+
+class ValidatorElectionVote(Transaction):
+
+    VALIDATOR_ELECTION_VOTE = 'VALIDATOR_ELECTION_VOTE'
+    # NOTE: This class inherits TRANSFER txn type. The `TRANSFER` property is
+    # overriden to re-use methods from parent class
+    TRANSFER = VALIDATOR_ELECTION_VOTE
+    ALLOWED_OPERATIONS = (VALIDATOR_ELECTION_VOTE,)
+
+    def validate(self, bigchain, current_transactions=[]):
+        """Validate election vote transaction
+        NOTE: There are no additional validity conditions on casting votes i.e.
+        a vote is just a valid TRANFER transaction
+
+        For more details refer BEP-21: https://github.com/bigchaindb/BEPs/tree/master/21
+
+        Args:
+            bigchain (BigchainDB): an instantiated bigchaindb.lib.BigchainDB object.
+
+        Returns:
+            `True` if the election vote is valid
+
+        Raises:
+            ValidationError: If the election vote is invalid
+        """
+        self.validate_transfer_inputs(bigchain, current_transactions)
+        return self
+
+    @classmethod
+    def to_public_key(cls, election_id):
+        return base58.b58encode(bytes.fromhex(election_id))
+
+    @classmethod
+    def generate(cls, inputs, recipients, election_id, metadata=None):
+        (inputs, outputs) = cls.validate_transfer(inputs, recipients, election_id, metadata)
+        election_vote = cls(cls.VALIDATOR_ELECTION_VOTE, {'id': election_id}, inputs, outputs, metadata)
+        cls.validate_schema(election_vote.to_dict(), skip_id=True)
```

Comment:
Why do you skip `id` validation here?

Reply:
The id cannot be validated because the transaction hasn't been signed yet and its only after signature we calculate the id of the transaction (the explaination is same to what we discussed for this [one](https://github.com/bigchaindb/bigchaindb/blob/master/bigchaindb/upsert_validator/validator_election.py#L123))

Reply:
True, thanks for the clarification!